design/thin-edge-actors-design.md

```diff
@@ -553,23 +583,273 @@ impl From<Msg2> for Msg {
 
 #### Channel types
 
-#### Channel creation and ownership
+We don't want to expose that the messages are sent over `mpsc` channels. 
+Abstracting both the senders and receivers, we can implement specific mechanisms
+to manage message priority, timeout, cancellation and more.
 
-#### Actor with no inputs
+The receiver part is abstracted by a concrete type: a mailbox for a specific type of messages.
+A key design point is that *all* the interactions with an actor must go through such a mailbox,
+including runtime errors, cancellations and timeouts.
+Doing so simplifies the actor event loops to `while let Some(message) = self.mailbox.next().await { ... }`
+and gives the flexibility to improve the system with new message delivery mechanisms.
 
-#### Out-of-band runtime messages
+```rust
+/// A mailbox that gather *all* the messages sent to an actor, including runtime messages.
+impl<M : From<RuntimeMessage>> Mailbox<M> {
+  /// Pop from the mailbox the message with the highest priority
+  /// 
+  /// Await till a messages is available.
+  /// Return `None` when all the senders to this mailbox have been dropped and all the messages consumed.
+  pub async fn next(&mut self) -> Option<M> {
+    // ...
+  }
+  
+  /// Turn on/off logging of the messages consumed from this mailbox.
+  /// 
+  /// The messages are logged when returned by the `next()` method.
+  pub fn log_inputs(&mut self, on: bool) {
```

Comment:
The developer is expected to turn this logging ON/OFF in code? Or this API will be connected to the external logging system like the `tracing` logger to control it more dynamically?

Reply:
This is a sketch of the API. Logging will be turn on/off by the user.

What I'm doing with this document is to find the right owner for each feature. Giving the responsibility of message logging to the mailbox makes this feature available to all actors without having to change the actor code if we improve logging.